### PR TITLE
python

### PR DESCRIPTION
**Pull request by MeisiSaalim**

### CODE REVIEW

File: Assignment4/MeisiSaalim-0X9230823/task07.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+for subClass, _, _ in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
+    print(subClass)
+
+print("SPARQL: ")
+
+q1 = prepareQuery(
+        """
+            SELECT ?subClass
+            WHERE{
+                ?subClass rdfs:subClassOf ns:LivingThing .
```
Comment (dgarijo) [inline Assignment4/MeisiSaalim-0X9230823/task07.py:40]: Does not retrieve subclasses of subclasses of living thing. Use `*`

File: Assignment4/MeisiSaalim-0X9230823/task07.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+ns = Namespace("http://somewhere#")
+
+print("RDFLib: ")
+
+for subClass, _, _ in g.triples((None, RDFS.subClassOf, ns.LivingThing)):
```
Comment (dgarijo) [inline Assignment4/MeisiSaalim-0X9230823/task07.py:31]: Same issue as below. Use a recursive function

File: Assignment4/MeisiSaalim-0X9230823/task07.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    ?entity foaf:knows ?person1 .
+    ?entity foaf:knows ?person2 .
+    FILTER (?person1 != ?person2)
+  }
+  GROUP BY ?entity
+  HAVING (COUNT(?person1) >= 2)
+  """,
+  initNs = {"rdfs":RDFS, "ns":ns, "foaf":FOAF}
+)
+
```
Comment (dgarijo) [inline Assignment4/MeisiSaalim-0X9230823/task07.py:169]: Several exercises missing in rdflib!

**Review by dgarijo — CHANGES_REQUESTED** [review]
Please address remaining issues

**Comment by anaigmo** [comment]
Submitted and mostly fixed, no more PRs allowed

**Pull request merged**